### PR TITLE
Add convenience method

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java [hunk 1]
```diff
@@ -36,6 +41,23 @@ public class DiagMessage {
...
+
+  /**
+   * Create a DiagMessage.
+   *
+   * @param source where to report the diagnostic
+   * @param kind the kind of message
+   * @param messageKey the message key
+   * @param args the arguments that will be interpolated into the localized message
+   */
+  public DiagMessage(
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java:55]: This is never called with a non-null `source` argument.  What is you plan for using it?

**Review by smillst — CHANGES_REQUESTED** [review]

File: framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java (hunk 1 above)
Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java:55]: Previously, #5862 used it, but that is no longer the case.  I can remove this feature if you like -- let me know your preference.

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java:55]: I'd rather not add code that isn't used.

Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java]: This pull request no longer makes important changes.  It just adds one convenience method.  I'm happy to merge it or discard it -- whichever you prefer.

**Review by smillst — APPROVED** [review]

**Comment by mernst** [comment]
@smillst This is ready for re-review.

**Pull request merged**